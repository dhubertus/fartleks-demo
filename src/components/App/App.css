nav {
  height: 12vh;
  width: 100%;
  display: flex;
  align-items: center;
  background: #3d3e42;
  position: fixed;
  top: 0;
  z-index: 300;
  box-shadow: 0px 1px 1px black;
}

nav h1 {
  font-size: 2.3em;
  color: #00aced;
  font-weight: 100;
  margin-left: 15px;
}

#logo {
  background: url('../../assets/icons/running.svg');
  height: 35px;
  width: 35px;
  background-repeat: no-repeat;
  background-size: cover;
  margin-left: 50px;
}

footer {
  height: 40px;
  width: 100%;
  background: #3d3e42;
  margin-top: 30px;
}

#nav-spacer {
  height: 12vh;
  width: 100%;
}

#split {
  display: flex;
  height: 13vh;
  width: 100%;
  justify-content: center;
}

#arrow {
  background: url('../../assets/icons/down-arrow.svg');
  height: 30px;
  width: 30px;
  background-repeat: no-repeat;
  background-size: cover;
  align-self: center;
  margin: 0 15px;
}

#arrow:hover {
  cursor: pointer;
}

.split-line {
  height: 6.5vh;
  width: 42%;
  border-bottom: 1px solid #3d3e42;
}

#graph-title,
#graph-use {
  text-align: center;
  font-weight: 100;
  color: #3d3e42;
}
