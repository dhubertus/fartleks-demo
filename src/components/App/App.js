import React, { Component } from 'react';
import sampleData from '../../test-data/workout-data.json';

import HandleDataApi from '../../helpers/HandleDataApi';
import RouteMap from '../RouteMap/RouteMap';
import Graph from '../Graph/Graph';
import EffortDisplay from '../EffortDisplay/EffortDisplay';
import BestEffortsContainer from '../../containers/BestEffortsContainer';


class App extends Component {
  constructor() {
    super();
    this.dataApi = new HandleDataApi();
  }

  componentWillMount() {
    const cleanData = sampleData.samples.filter((item) => {
      if ((typeof item.values.power) === 'number') {
        return item;
      }
    });

    const twenty = this.dataApi.calculateBestEffort(cleanData, 1200, 'twenty');
    const fifteen = this.dataApi.calculateBestEffort(cleanData, 900, 'fifteen');
    const ten = this.dataApi.calculateBestEffort(cleanData, 600, 'ten');
    const five = this.dataApi.calculateBestEffort(cleanData, 300, 'five');
    const one = this.dataApi.calculateBestEffort(cleanData, 60, 'one');

    const bestEfforts = {
      one,
      five,
      ten,
      fifteen,
      twenty,
    };

    this.props.handleBestEfforts(bestEfforts);
  }

  render() {
    return (
      <div className="App">
        <nav>
          <div id="logo"></div>
          <h1>Fartleks</h1>
        </nav>
        <div id="nav-spacer"></div>
        <RouteMap />
        <div id="split">
          <div class="split-line"></div>
          <div id="arrow"></div>
          <div class="split-line"></div>
        </div>
        <EffortDisplay />
        <h3 id="graph-title">Power vs Time</h3>
        <h5 id="graph-use">Select two points to trace data on map</h5>
        <Graph />
        <footer></footer>
      </div>
    );
  }
}

export default BestEffortsContainer(App);
