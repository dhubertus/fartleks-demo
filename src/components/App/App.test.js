import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
// import { Provider } from 'react-redux';
// import BestEffortsContainer from '../../containers/BestEffortsContainer';
import configureMockStore from 'redux-mock-store';

import App from './App';

const mockStore = configureMockStore()({
  bestEfforts: {
    one: [
      [1, 2],
      [2, 3],
    ],
    five: [
      [1, 2],
      [2, 3],
    ],
    ten: [
      [1, 2],
      [2, 3],
    ],
    fifteen: [
      [1, 2],
      [2, 3],
    ],
    twenty: [
      [1, 2],
      [2, 3],
    ],
  },
  selectedData: [],
});

const wrapper = mount(<Provider store={mockStore}><App /></Provider>);
const Component = wrapper.find(App);


// Enzyme.configure({ adapter: new Adapter() });

describe('App', () => {
  it('renders without crashing', () => {
    console.log(Component);
    // expect(wrapper.find('.App').length).toEqual(1);
  });
});
