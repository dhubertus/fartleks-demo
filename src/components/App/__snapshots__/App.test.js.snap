// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Effort Display Tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    handleBestEfforts={[Function]}
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <nav
          id="page-nav"
>
          <div
                    id="logo"
          />
          <h1>
                    Fartleks
          </h1>
</nav>,
        <div
          id="nav-spacer"
/>,
        <Connect(RouteMap) />,
        <div
          id="split"
>
          <div
                    className="split-line"
          />
          <div
                    id="arrow"
                    onClick={[Function]}
          />
          <div
                    className="split-line"
          />
</div>,
        <Connect(EffortDisplay) />,
        <h3
          id="graph-title"
>
          Power vs Time
</h3>,
        <h5
          id="graph-use"
>
          Select Range Of Data To Display On Map
</h5>,
        <Connect(Graph) />,
        <footer />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              id="logo"
/>,
            <h1>
              Fartleks
</h1>,
          ],
          "id": "page-nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "logo",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Fartleks",
            },
            "ref": null,
            "rendered": "Fartleks",
            "type": "h1",
          },
        ],
        "type": "nav",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "nav-spacer",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="split-line"
/>,
            <div
              id="arrow"
              onClick={[Function]}
/>,
            <div
              className="split-line"
/>,
          ],
          "id": "split",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "split-line",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "arrow",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "split-line",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Power vs Time",
          "id": "graph-title",
        },
        "ref": null,
        "rendered": "Power vs Time",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select Range Of Data To Display On Map",
          "id": "graph-use",
        },
        "ref": null,
        "rendered": "Select Range Of Data To Display On Map",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "footer",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <nav
            id="page-nav"
>
            <div
                        id="logo"
            />
            <h1>
                        Fartleks
            </h1>
</nav>,
          <div
            id="nav-spacer"
/>,
          <Connect(RouteMap) />,
          <div
            id="split"
>
            <div
                        className="split-line"
            />
            <div
                        id="arrow"
                        onClick={[Function]}
            />
            <div
                        className="split-line"
            />
</div>,
          <Connect(EffortDisplay) />,
          <h3
            id="graph-title"
>
            Power vs Time
</h3>,
          <h5
            id="graph-use"
>
            Select Range Of Data To Display On Map
</h5>,
          <Connect(Graph) />,
          <footer />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                id="logo"
/>,
              <h1>
                Fartleks
</h1>,
            ],
            "id": "page-nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "logo",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Fartleks",
              },
              "ref": null,
              "rendered": "Fartleks",
              "type": "h1",
            },
          ],
          "type": "nav",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "nav-spacer",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="split-line"
/>,
              <div
                id="arrow"
                onClick={[Function]}
/>,
              <div
                className="split-line"
/>,
            ],
            "id": "split",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "split-line",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "arrow",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "split-line",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Power vs Time",
            "id": "graph-title",
          },
          "ref": null,
          "rendered": "Power vs Time",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select Range Of Data To Display On Map",
            "id": "graph-use",
          },
          "ref": null,
          "rendered": "Select Range Of Data To Display On Map",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "footer",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
